### ISSUE TITLE
Define z-index hierarchy for React components

### ISSUE THREAD

**Issue by mimarz** [digdir/designsystemet#500]
Its common for component libraries to define a z-index hierarchy to make sure all floating elements of components work as intended and don't interfere with one and another. 

MUI has a good write explanation for this and I suggest use their hierarchy as start. 

Notes:
- https://mui.com/material-ui/customization/z-index/

Todo:
- Make a list based on MUI's z-index hierarchy for our components.
  - `Select` would for example end up on 1500
  
  
Suggest we just follow mui's table for now.

Our equivalant would be:

```
sidebar: 1200
modal: 1300
tooltip/popover/dropdownmenu: 1500
```

**Comment by TomasEng** [comment]
I suggest creating a file where we list all default z-index values in variables. In that way, we can easily see the hierarchy and make changes if necessary. What do you think, @mimarz?

**Comment by mimarz** [comment]
Sure, Its not something you change that often, but it would make it configurable/overridable for our users. Would have to find good names for each "tier". 

Now where do we put it 🤔 Best spot would maybe be to put this in the generated token.css we have for each brand.

**Comment by TomasEng** [comment]
Alternatively we could put it in a separate file somewhere in the React package (maybe a new `style` directory for CSS utils), so we don't mix them with the Figma generated tokens. I think I would prefer naming the variables directly after the components they are intended for, in order to minimize complexity.

**Comment by mimarz** [comment]
Yeah, I agree on the naming. 

A bit unsure about having them in another file as it could maybe be prone to human error? Since it would be css variables, you risk people not importing/including the correct file(s), or just the fact that we need to remember in which "set" we have the different css vars 🤔 

Just feel it might be cleaner and less prone for errors to have all css vars used by the library in one single file.

**Comment by TomasEng** [comment]
Good point. I agree if it's not too complicated to achieve regarding the Figma generation script.

**Comment by mimarz** [comment]
Its pretty straight forward to add it to the existing script. Just need to decide where in the transformation we want to add it in terms of if we want it to be presented in the storefront or not.

**Comment by mimarz** [comment]
This will be looked at when we start on #82, #219

**Comment by Barsnes** [comment]
Closing this since we now use native popover and dialog, making this issue not relevant
